src/sidebar/components/TopBar.tsx

```diff
@@ -121,17 +105,7 @@ function TopBar({
         <div className="grow flex items-center justify-end">
           {isSidebar && (
             <>
-              {pendingUpdateCount > 0 && (
-                <IconButton
-                  icon={RefreshIcon}
-                  onClick={applyPendingUpdates}
-                  size="xs"
-                  variant="primary"
-                  title={`Show ${pendingUpdateCount} new/updated ${
-                    pendingUpdateCount === 1 ? 'annotation' : 'annotations'
-                  }`}
-                />
-              )}
+              <PendingUpdatesButton onClick={applyPendingUpdates} />
```

Comment:
Given that `PendingUpdatesButton` is already pretty self sufficient and makes use of the `toastMessengerService`, I wonder if you might go ahead and inject the `streamer` service into it, and free `TopBar` from having to know about `streamer`. Thoughts?

Reply:
Good point!